src/ScheduleFunctions.cpp

```diff
@@ -1922,6 +1924,29 @@ bool validate_schedule(Function f, Stmt s, const Target &target, bool is_output,
                            << target.to_string() << "\n";
             }
         }
+        if (s.allow_race_conditions()) {
+            allow_race_conditions_count++;
+        }
+        for (const auto &split : s.splits()) {
+            if (split.tail == TailStrategy::ShiftInwards) {
```

Comment:
I think this warning needs to be way more selective to be useful. Nearly all Halide programs use a shiftinwards somewhere (vectorize(x, 8)), but usually there's no outer parallel loop (so no race condition). The sniffer really needs to see if the shiftinwards split has an outer var that is marked parallel, or if the outer var is split again, then if any of its children are eventually marked parallel.

Reply:
True. PTAL.